tests/Avalonia.Controls.UnitTests/Presenters/ContentPresenterTests_Standalone.cs

```diff
@@ -264,5 +264,118 @@ public void Should_Reset_InheritanceParent_When_Child_Removed()
             // InheritanceParent is exposed via StylingParent.
             Assert.Same(logicalParent, ((IStyledElement)child).StylingParent);
         }
+
+        [Fact]
+        public void Should_Create_Child_Even_With_Null_Content_When_ContentTemplate_Is_Set()
+        {
+            var target = new ContentPresenter
+            {
+                ContentTemplate = new FuncDataTemplate<object>(_ => true, (_, _) => new TextBlock
+                {
+                    Text = "Hello World"
+                }),
+                Content = null
+            };
+
+            target.UpdateChild();
+
+            var textBlock = Assert.IsType<TextBlock>(target.Child);
+            Assert.Equal("Hello World", textBlock.Text);
+        }
+
+        [Fact]
+        public void Should_Create_Child_Even_With_Null_Content_When_DataTemplates_Is_Set()
+        {
+            var target = new ContentPresenter
+            {
+                DataTemplates =
+                {
+                    new FuncDataTemplate<object>(_ => true, (_, _) => new TextBlock
+                    {
+                        Text = "Hello World"
+                    })
+                },
+                Content = null
+            };
+
+            target.UpdateChild();
+
+            var textBlock = Assert.IsType<TextBlock>(target.Child);
+            Assert.Equal("Hello World", textBlock.Text);
+        }
+
+        [Fact]
+        public void Should_Create_Child_Even_With_Null_Content_When_DataTemplates_Is_Set_On_Parent()
```

Comment:
We probably don't want to allow this behavior to avoid possible problems. See "breaking changes" section.

Reply:
Aaaaand we have it 
![image](https://user-images.githubusercontent.com/3163374/125176702-8e678580-e1a3-11eb-967d-7db79f211a6b.png)

Reply:
Updated PR, not ContentPresenter does not create Child, if only DataTemplates was set neither on the control or on the parent. User needs to use ContentPresenter explicitly to avoid unexpected problems.